### PR TITLE
unregisterOutput() + isPinOrModeChanged() + Stepper::stepPin+enablePin + Kinetis fix

### PR DESCRIPTION
**Pull request by andreika-git**
Refactoring and fixes around EFI_ACTIVE_CONFIGURATION_IN_FLASH.
Also refactor Stepper pins and fix stepper fault if the enable pin was not set.
Also fix EFI_ACTIVE_CONFIGURATION_IN_FLASH for Kinetis.

Tested on Kinetis.

### CODE REVIEW

**Comment by rusefi** [comment]
Can one of the admins verify this patch? (meaning the author is not on the white list)

**Comment by rusefi** [comment]
Can one of the admins verify this patch? (meaning the author is not on the white list)

**Comment by rusefi** [comment]
Can one of the admins verify this patch? (meaning the author is not on the white list)

**Comment by rusefi** [comment]
Can one of the admins verify this patch? (meaning the author is not on the white list)

File: firmware/controllers/injector_central.cpp [hunk 1]
```diff
@@ -250,16 +250,6 @@ class BenchController : public PeriodicController<UTILITY_THREAD_STACK_SIZE> {
 
 static BenchController instance;
 
-void OutputPin::unregisterOutput(brain_pin_e oldPin, brain_pin_e newPin) {
```
Comment (andreika-git) [inline firmware/controllers/injector_central.cpp]: Moved to efi_gpio.cpp

**Comment by rusefi** [comment]
add to while list

**Pull request merged**